### PR TITLE
fix(auth): support regional Google domains for SID cookie extraction

### LINKED ISSUE #20: notebooklm list 命令缺少 SID cookie 导致认证失败。
运行 notebooklm list 命令时，提示 Missing required cookies: {'SID'} 错误，即使已经通过 notebooklm login 成功登录。

  复现步骤

  1. 运行 notebooklm login，在浏览器中完成 Google 登录
  2. 等待 NotebookLM 首页完全加载
  3. 按 ENTER 保存会话
  4. 运行 notebooklm list
  5. 收到错误：Error: Missing required cookies: {'SID'}

  诊断信息

  检查 ~/.notebooklm/storage_state.json，发现：
  - notebooklm.google.com 域有 OSID 和 __Secure-OSID cookie ✅
  - notebooklm.google.com 域缺少 SID cookie ❌
  - .youtube.com 和 .google.com.sg 域有 SID cookie

  有趣的是：
  - notebooklm status 命令可以正常工作
  - 只有 notebooklm list 命令失败

  环境信息

  - 操作系统：macOS (Darwin 25.2.0)
  - notebooklm-py 版本：0.1.1
  - Python：3.9
  - 安装方式：pip install notebooklm-py[browser]

  可能的原因

  登录过程的 cookie 捕获机制没有正确获取 notebooklm.google.com 域的 SID cookie，只捕获了跨域的 SID cookie。

### PR DESCRIPTION
**Pull request by teng-lin**
## Summary
- Fix "Missing required cookies: {'SID'}" error for users in regional Google domains
- Add `GOOGLE_REGIONAL_CCTLDS` whitelist with 70+ regional domains
- Create unified `_is_google_domain()` function for consistent validation
- Update both `_is_allowed_auth_domain` and `_is_allowed_cookie_domain`

Fixes #20

## Regional Domain Support

| Pattern | Examples | Countries |
|---------|----------|-----------|
| `.google.com.XX` | .google.com.sg, .google.com.au | Singapore, Australia, Brazil, etc. |
| `.google.co.XX` | .google.co.uk, .google.co.jp | UK, Japan, India, Korea, etc. |
| `.google.XX` | .google.de, .google.fr | Germany, France, Italy, Spain, etc. |

## Security

Uses whitelist approach (safer than regex):
- Only explicit domains in `GOOGLE_REGIONAL_CCTLDS` accepted
- Leading dot required (`.google.com` not `google.com`)
- Case-sensitive matching (browsers normalize to lowercase)
- No subdomain leakage (`evil.google.de` rejected)

## Test plan
- [x] 128 unit tests for auth module (+65 new tests)
- [x] Parametrized tests for all 70+ regional domains
- [x] Edge case tests (uppercase, whitespace, malformed, suffix exploits)
- [x] Integration tests for cookie extraction and httpx cookies
- [x] All 1101 tests passing

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: src/notebooklm/auth.py [hunk 1]
```diff
@@ -110,12 +196,69 @@ async def from_storage(cls, path: Path | None = None) -> "AuthTokens":
...
+
+    Returns:
+        True if domain is allowed for auth cookies.
+    """
+    # Exact match against primary allowlist (includes notebooklm.google.com, etc.)
+    if domain in ALLOWED_COOKIE_DOMAINS:
+        return True
+
+    # Check if it's a valid Google domain (base or regional)
+    return _is_google_domain(domain)
```
[the remark this answers is not in the document]
Comment (teng-lin) [reply src/notebooklm/auth.py]: Addressed in commit 59df9c0: Simplified to single return statement combining the allowlist check with _is_google_domain().
